pkg/agent/protocol/http/proxy.go

```diff
@@ -211,3 +212,14 @@ func (p *proxy) Force() error {
 	}
 	return nil
 }
+
+// supportedMetrics is a helper function that returns the metrics that the http proxy supports and thus should be
+// pre-initialized to zero. This function exists as a workaround, as currently the handler tests need to initialize the
+// MetricMap. It should not be needed when metric tests are moved from the handler to the proxy.
+func supportedMetrics() []string {
```

Comment:
I do not agree with this comment. Moreover, I think this method should be public as part of the contract of the Proxy.

Reply:
I don't see why having the list of metrics available should change the code for checking, therefore I don't think the "verbosity" argument is valid.

In any case, my point was that having the proxy to know the supported list of metrics is not in my opinion a hack around the limitations of testing but something we expect it to know. The hack, in any case, is testing this at the handler level and not the proxy level. Maybe this should be stated more clearly in the comment.

Regarding exposing it or not in the API, it is secondary. But I think it is valid because it is not immediately evident that the `proxy.Metrics()` method will return all supported metrics regardless of whether they have a non-zero value or not. We should make this more explicit in the documentation:

```
// Metrics returns a map of counter-type metrics. Implementations may return zero or more of the metrics defined
// below, as well as any number of implementation-defined metrics.
```

Reply:
These are good points, I've clarified both comments to state more clearly how supported metrics are exposed and the purpose of `supportedMetrics`. LMK if these look better to you 🙂